src/Utils/LogRanker.ts

```diff
@@ -0,0 +1,132 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.  
+ * Licensed under the MIT License.
+ */
+import * as CLM from '@conversationlearner/models'
+import { TreeUtils } from './TreeUtils'
+
+export interface LogScore {
```

Comment:
Please fix: these fields could benefit from some comments.  For example, "knownPath" is comprehensible to people who have been in design discussions on this, but probably obtuse to somebody new to the project.